docs/proposal/2022-09-28-event-manager-extensibility.md

```diff
@@ -0,0 +1,302 @@
+event payload and process it in Botkube. This is a typical example to show how we can have a dynamic handling mechanism for Source Plugins.
+
+To sum up source plugin system;
+- They can consume external system like Kubernetes events
+- They can be in noop format, and they would have only plugin configuration to register that in Botkube
+- They can also spin up an handler to handle custom events from other sources like Prometheus
+
+
+### Executor Plugin Structure
+Executor plugins would have nearly same flow as Source Plugins with a small difference. Executor plugins would be triggered
+whenever Botkube receives a command from end-user. So, once Botkube receives command, based on the command prefix it can 
+resolve which plugin to call and uses response on gRPC execution. Think about `KubectlExecutor`, once Botkube gets `kubectl get pods`
+it can resolve plugin by its prefix which is `kubectl` and call `KubectlExecutor` gRPC endpoint automatically and parses the response.
+As you can also see, Botkube does not need to know how `kubectl` works, it simply delegates all the responsibility to specific plugin.
+
+![](./assets/kubectl-executor-plugin.png)
+
+### Cloud Events
+Cloudevents schema has a proven standard for the cloud native industry where well-known products uses this schema to distribute their events
+via this contract. The main take of this, you don't need to re-invent wheel from scratch while you are designing your data format. In Botkube,
+we can also use [one of the Cloudevents SDK](https://github.com/cloudevents/sdk-go) to spin up an handler inside Botkube so that any cloudevents 
+compatible source can send data to Botkube. The main example to this is [Keptn](https://keptn.sh/), an open source application life-cycle orchestration tool, 
+they send events in cloudevents format. If we want to ship Keptn events to Botkube to be able to send notifications, we may need to add new integration to Keptn
+so that it will send events to Botkube `/events` endpoint which handles cloudevents.
+
+#### Botkube Cloudevents Data Structure
+You will see very simple suggested event format for Botkube as shown below.
+```json
+{
+  "id" : "e2361318-e50b-472e-8b17-13dbac1daac1",   
+  "source" : "Kubernetes",     
+  "specversion" : "1.0",                             
+  "time" : "2022-03-23T14:35:39.738Z",               
+  "data" : {                                         
+    "namespace" : "Botkube",
+    "deployment" : {
+      "name" : "test",
+      "replicas" : 5
+    }
+  }
+}
+```
+
+`id`: This is the unique identifier for the event, and this should be decided by end users. UUID is not a mandatory stuff, it can
+be also ARN for example `ec2:myinstance:12345`.
+
+`source`: This refers to plugin name. Once Botkube receives and event on `/events` endpoint, `source` field would be validated
+to see we already have a registered plugin for this event or not. 
+
+`specversion`: To have proper versioning for Botkube event format.
```

Comment:
```suggestion
`specVersion`: To have proper versioning for Botkube event format.
```
I don't see a reason to not use the camel case here.

Reply:
I just wanted to follow conventions as you can see an example [here](https://github.com/keptn/spec/blob/master/cloudevents.md), but we can try using camelcase and see if cloudevent SDK works.